chemprop/featurizers/molecule.py

```diff
@@ -41,3 +41,24 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        pass
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        if mol.GetNumHeavyAtoms() == 0:
+            features = np.array(
+                [0 if name == "SPS" else func(mol) for name, func in sorted(Descriptors.descList)]
+            )
+        else:
+            features = np.array([func(mol) for name, func in sorted(Descriptors.descList)])
+
+        NAN_TOKEN = 0
+        features[np.isnan(self.x_d)] = NAN_TOKEN
```

Comment:
self.x_d is molecular descriptor for a data point

Reply:
I see, but if it is an attribute in the data points, can it be accessed here? Besides, it seems this code [here](https://github.com/chemprop/chemprop/blob/main/chemprop/data/datapoints.py#L43-L45) already handled the issues about nan values.

Reply:
I had copied this code (but accidentally left self.x_d) because some of the rdkit2d features would return nan (https://github.com/rdkit/rdkit/discussions/5824) and I feel it is the responsibility of the featurizer to not return nans. 

But in any event, these features that would return nan are not calculated by descriptastorus. There are 12 features that rdkit descriptors have that Descriptastorus doesn't. But I figure the consistency with chemprop v1 is worth not having these extra features especially since a user can still calculate them manually.